sdk/lib/opentelemetry/sdk/internal.rb

```diff
@@ -20,8 +20,12 @@ def valid_key?(key)
         key.instance_of?(String)
       end
 
+      def valid_numeric?(value)
+        value.is_a?(Integer) || value.is_a?(Float)
+      end
+
       def valid_simple_value?(value)
-        value.instance_of?(String) || value == false || value == true || value.is_a?(Numeric)
+        value.instance_of?(String) || value == false || value == true || valid_numeric?(value)
```

Comment:
Further to the `instance_of?` vs `is_a?` question, we are using `==` here for booleans, which *may* be too loose a constraint. We also have `boolean?` defined above in terms of `is_a?` and used in `valid_array_value?` below. Should we instead use `instance_of?` in `boolean?` and use `boolean?(value)` here?